packages/flame/test/components/advanced_button_component_test.dart

```diff
@@ -0,0 +1,220 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/src/events/flame_game_mixins/multi_tap_dispatcher.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('AdvancedButtonComponent', () {
+    testGolden(
+      'label renders correctly',
+      (game) async {
+        await game.add(
+          AdvancedButtonComponent(
+            defaultSkin: RectangleComponent(size: Vector2(100, 40)),
+            defaultLabel: RectangleComponent(
+              size: Vector2(20, 10),
+              paint: Paint()..color = const Color(0xFFFF0000),
+            ),
+          ),
+        );
+      },
+      goldenFile: '../_goldens/advanced_button_component.png',
+    );
```

Comment:
The size is huge compared to the area you are rendering on, you can pass in the `size` argument to `testGolden` to make it smaller.

Reply:
```
          AdvancedButtonComponent(
            defaultSkin: RectangleComponent(size: Vector2(20, 10)),
            defaultLabel: RectangleComponent(
              size: Vector2(5, 2),
              paint: Paint()..color = const Color(0xFFFF0000),
            ),
          )
```

Did I understand correctly?

Reply:
For some reason my commit is not sent

Reply:
You can add a size to the `testGolden` method that you are using, right above `goldenFile` for example, to set the size of the canvas that it should render on.

Reply:
> For some reason my commit is not sent

You have to pull first, since I pushed updated goldens to your branch.

Reply:
I got it, ready!